kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h

```diff
@@ -247,6 +247,26 @@ class ResidualBasedNewtonRaphsonStrategy
      */
     ~ResidualBasedNewtonRaphsonStrategy() override
     {
+        // If the linear solver has not been deallocated, clean it before
+        // deallocating mpA. This prevents a memory error with the the ML
+        // solver (which holds a reference to it).
+        auto p_linear_solver = GetBuilderAndSolver()->GetLinearSystemSolver();
```

Comment:
Some of these lines are repeated in `Clear()`

```suggestion
        Clear();
        mpA.reset();
        mpDx.reset();
        mpb.reset();
```

Reply:
I know my solution is not the most elegant, but I need to reset the pointers before calling `Clear()`. Essentially, the point is that `TrilinosSpace::Clear()` does things that are not allowed after calling `MPI_Finalize` (which I have no way to control has not happened before this point).

Reply:
ok, then I think you don't need to call `Clear` in the destructor for the ML problem since you anyway clear the linear solver.